app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,114 @@
+class MerchantsController < ApplicationController
+  before_action :find_merchant, only: [:show, :edit, :update, :destroy, :fulfillment, :show_order, :ship_orderitems]
+  before_action :account_owner?, only: [:show, :edit, :update, :destroy, :fulfillment, :show_order, :ship_orderitems]
+
+  def new
+  end
+
+  def show_order
+    @order = Order.find_by(id: params[:order_id])
+    if @order == nil
+      flash[:status] = :error
+      flash[:result_text] = "Invalid order"
+      return redirect_back(fallback_location: root_path)
+    end
+    @title = "Review Order #{@order.id}"
+    @orderitems = @order.orderitems.select { |item| item.product.merchant_id == @merchant.id }
+    @unshipped = (@orderitems.select { |item| item.status != "shipped"}.length >0 )
+  end
+
+  def fulfillment
+    if params[:id].to_i == session[:merchant_id]
+      status = params[:status]
+      orderitems = @merchant.join_orderitems(status)
+      @orders = {}
+      orderitems.each do |orderitem|
+        if @orders[orderitem.order_id.to_s] == nil
+          new_revenue = orderitem.quantity * orderitem.product.price
+          new_items = [orderitem]
+        else
+          new_revenue = @orders[orderitem.order_id.to_s][:revenue] + (orderitem.quantity * orderitem.product.price)
+          new_items = @orders[orderitem.order_id.to_s][:items]
+          new_items << orderitem
+        end
+        @orders[orderitem.order_id.to_s] = {status: orderitem.order.status, purchase_date: orderitem.order.purchase_datetime,revenue: new_revenue, items: new_items, unshipped: ""}
+      end
+      @orders.each do |order_id, hash|
+        unshipped = hash[:items].select { |item| item.status != "shipped"}.length > 0
+        @orders[order_id][:unshipped] = unshipped
+      end
+      @total_revenue = orderitems.sum {|orderitem| ( (orderitem.order.status == "cancelled") ? 0 : (orderitem.quantity * orderitem.product.price) ) }
+      @statuses = ["paid", "complete", "cancelled"]
+      @revenue_by_status = {}
+      @statuses.each do |a_status|
+        @revenue_by_status[a_status] = orderitems.sum {|orderitem| orderitem.order.status == a_status ? (orderitem.quantity * orderitem.product.price) : 0}
+      end
+      @title = "Fulfillment"
+    else
```

Comment:
this if/else case is redundant with your other controller filters; it will never hit this "else" block because it already checks if you're the right merchant logged in with `find_merchant` and `account_owner?`